crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2636,7 +2660,11 @@ impl<'db> Class<'db> {
     pub fn is_subclass_of(self, db: &'db dyn Db, other: Class) -> bool {
         // `is_subclass_of` is checking the subtype relation, in which gradual types do not
         // participate, so we should not return `True` if we find `Any/Unknown` in the MRO.
-        self.iter_mro(db).contains(&ClassBase::Class(other))
+        self.is_subclass_of_base(db, other)
+    }
+
+    fn is_subclass_of_base<C: Into<ClassBase<'db>>>(self, db: &'db dyn Db, other: C) -> bool {
+        self.iter_mro(db).contains(&other.into())
```

Comment:
This seems likely useful in a future PR when we do want to implement some of the special handling for classes that inherit `Any`, I'd leave it!